src/jdk.jpackage/share/classes/jdk/jpackage/internal/IOUtils.java

```diff
@@ -56,6 +56,8 @@
 public class IOUtils {
 
     public static void deleteRecursive(Path directory) throws IOException {
+        final IOException [] exception = { (IOException) null };
```

Comment:
Do we know `Files.walkFileTree()` synchronizes calls on callback object? If not, I'd use `AtomicReference` to store the first exception.

Reply:
That seems like overkill.  walkFileTree must call visitFile, preVisitDirectory, and postVisitDirectory synchronously, because their return value tells walkFileTree where to go next.

Reply:
I can use AtomicReference instead of Array to hold the IOException, but must I lock around access, there is no "setIfNull()" method

Reply:
Actually there is: compareAndSet(newValue, null)